src/debug.c

```diff
@@ -2007,43 +2007,33 @@ void watchdogScheduleSignal(int period) {
     it.it_interval.tv_usec = 0;
     setitimer(ITIMER_REAL, &it, NULL);
 }
+void applyWatchdogPeriod() {
+    struct sigaction act;
 
-/* Enable the software watchdog with the specified period in milliseconds. */
-void enableWatchdog(int period) {
-    int min_period;
-
+    /* Disable watchdog when period is 0 */
     if (server.watchdog_period == 0) {
```

Comment:
Don't understand. I change the watchdog functions from two functions enable/disable to a single one. I'm not sure negating the `if` here would help much.

Reply:
the one for disable was below the one for enable, but in your new code it's the other way around, so maybe reversing that would help.
on the other hand, there are indentation changes too that would break this.
unless the enable code ends with a `return` so the disable code can remain unindented.

Reply:
I'll leave it as is. Since it's a completely new function I prefer to have written as clear as possible without blame considerations.